test/gateway_composer/accept_stake_request.js

```diff
@@ -12,22 +12,246 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+const BN = require('bn.js');
 const utils = require('../test_lib/utils');
 const { AccountProvider } = require('../test_lib/utils.js');
 const web3 = require('../test_lib/web3.js');
 
 const gatewayComposerUtils = require('./utils');
 
-contract('GatewayComposer::acceptStakeRequest', async () => {
-    contract('Positive Tests', async (accounts) => {
+contract('GatewayComposer::acceptStakeRequest', async (accounts) => {
+    describe('Negative Tests', async () => {
+        const accountProvider = new AccountProvider(accounts);
+
+        it('Fails when requestHash is invalid.', async () => {
```

Comment:
Global comment: "requestHash" -> "stakeRequestHash", for consistency

Suggest revising to, "Fails when stake request is not found", because that's what this test is actually checking for; in addition, an empty string is not _per se_ invalid and the tested function does not validate the content of the hash.

Reply:
```
 in addition, an empty string is not per se invalid and the tested function does not validate the content of the hash.
```
I didn't get this point ❓

Reply:
There are no validations in GC for a stakeRequestHash—GC is OK with an empty string as a stakeRequestHash. 

Additionally, this test confirms what happens when a matching stake request is not found, **not** what happens when a stakeRequestHash is invalid.

So:
- a test matching this test's description, "Fails when requestHash is invalid," would test functionality that doesn't exist; and
- this test does not match its description.